docs/localization.md

```diff
@@ -0,0 +1,103 @@
+# Localization
+
+During development, using [`NSLocalizedString()`](https://developer.apple.com/documentation/foundation/nslocalizedstring) in the code should be enough. You shouldn't need to touch the `Localizable.strings` files manually.
+
+During the release process, `NSLocalizedString` statements are scanned and stored in the `Localizable.strings` file. The file is then uploaded to [GlotPress](https://translate.wordpress.com/projects/woocommerce/woocommerce-ios/) for translation. Before the release build is finalized, all the translations are grabbed from GlotPress and saved back to the `Localizable.strings` files.
+
+## Always add Comments
+
+Always add a meaningful comment. If possible, describe where and how the string will be used. If there are placeholders, describe what each placeholder is. 
+
+```swift
+// Do
+let title = NSLocalizedString("Following %@",
+                              comment: "Title for a notice informing the user that they've successfully followed a site. %@ is a placeholder for the name of the site.")
+```
+
+```swift
+// Avoid
+let title = NSLocalizedString("Following %@", comment: "")
+```
+
+Comments help give more context to translators.
+
+## Do not use Variables
+
+Do not use variables as the argument of `NSLocalizedString()`. The string value will not be automatically picked up. 
+
+```swift
+// Do
+let myText = NSLocalizedString("This is the text I want to translate.", comment: "Put a meaningful comment here.")
+myTextLabel?.text = myText
+```
+
+```swift
+// Don't
+let myText = "This is the text I want to translate."
+myTextLabel?.text = NSLocalizedString(myText, comment: "Put a meaningful comment here.")
+```
+
+## Do not use Interpolated Strings
```

Comment:
Do you think we can add the use of positional placeholders for every variable in a `NSLocalizedString`, from @AliSoftware's p2 post pbMoDN-1Df-p2?

Reply:
+1 :) To be clear in case it wasn't, I'd suggest to have a dedicated `## Always use positional placeholders` section for that (not add a comment into an existing one)

Reply:
Thanks for the reminder, I added a section in 75d0233.